crates/fuel-core/src/service/genesis.rs

```diff
@@ -151,153 +146,146 @@ fn import_genesis_block(
-                    coin.tx_id.unwrap_or_else(|| {
-                        Bytes32::try_from(
-                            (0..(Bytes32::LEN - WORD_SIZE))
-                                .map(|_| 0u8)
-                                .chain(
-                                    (generated_output_index / 255)
-                                        .to_be_bytes()
-                                        .into_iter(),
-                                )
-                                .collect_vec()
-                                .as_slice(),
-                        )
-                        .expect("Incorrect genesis transaction id byte length")
-                    }),
-                    coin.output_index.unwrap_or_else(|| {
-                        generated_output_index = generated_output_index
-                            .checked_add(1)
-                            .expect("The maximum number of UTXOs supported in the genesis configuration has been exceeded.");
-                        (generated_output_index % 255) as u8
-                    }),
-                );
-
-                let coin = CompressedCoin {
-                    owner: coin.owner,
-                    amount: coin.amount,
-                    asset_id: coin.asset_id,
-                    maturity: coin.maturity.unwrap_or_default(),
-                    tx_pointer: TxPointer::new(
-                        coin.tx_pointer_block_height.unwrap_or_default(),
-                        coin.tx_pointer_tx_idx.unwrap_or_default(),
-                    ),
-                };
-
-                // ensure coin can't point to blocks in the future
-                if coin.tx_pointer.block_height() > state.height.unwrap_or_default() {
-                    return Err(anyhow!(
-                        "coin tx_pointer height cannot be greater than genesis block"
-                    ))
-                }
-
-                if db.storage::<Coins>().insert(&utxo_id, &coin)?.is_some() {
-                    return Err(anyhow!("Coin should not exist"))
-                }
-                coins_tree.push(coin.root()?.as_slice())
+
+    if let Some(coins) = &state.coins {
+        for coin in coins {
+            let utxo_id = UtxoId::new(
+                // generated transaction id([0..[out_index/255]])
+                coin.tx_id.unwrap_or_else(|| {
+                    Bytes32::try_from(
+                        (0..(Bytes32::LEN - WORD_SIZE))
+                            .map(|_| 0u8)
+                            .chain(
+                                (generated_output_index / 255).to_be_bytes().into_iter(),
+                            )
+                            .collect_vec()
+                            .as_slice(),
+                    )
+                    .expect("Incorrect genesis transaction id byte length")
+                }),
+                coin.output_index.unwrap_or_else(|| {
+                    generated_output_index += 1;
```

Comment:
It looks like you removed the `checked_add(1)` - is there any specific reason?

Reply:
Must have been an accident. Thanks!